File: tests/practice_run_tests/java/solution_files/Sol18.java
package tests.practice_run_tests.java.solution_files;
import java.util.List;

public class Sol18 {
  public static ListNode solve(List<Long> list) {
    if (list.isEmpty()) return null;
    ListNode ret = new ListNode(list.get(0), null);
    ListNode curr = ret;
    for (int i = 2; i < list.size(); i++) {
      curr.next = new ListNode(list.get(i), null);
      curr = curr.next;
    }
    return ret;
  }

  public static class ListNode {
    public long val;
    public ListNode next;

    public ListNode(long val, ListNode next) {
      this.val = val;
      this.next = next;
    }
  }
}

File: app_tests/python/solution_files/sol18.py
from __future__ import annotations 
from typing import Optional

class Solution:
  def solve(self, head: list[int]):
    if not head:
      return None
    entries = head[1:]
    ret = ListNode(entries[0])
    curr = ret
    for val in entries[1:]:
      curr.next = ListNode(val)
      curr = curr.next
    return ret
  
class ListNode:
  def __init__(self, val, next: Optional[ListNode] = None):
    self.val = val
    self.next = next